ignite/metrics/precision.py

```diff
@@ -15,28 +15,39 @@ class _BasePrecisionRecall(_BaseClassification):
     def __init__(
         self,
         output_transform: Callable = lambda x: x,
-        average: bool = False,
+        average: Union[bool, str] = False,
         is_multilabel: bool = False,
         device: Union[str, torch.device] = torch.device("cpu"),
     ):
 
+        if not (isinstance(average, bool) or average in ["micro", "weighted", "samples"]):
```

Comment:
When I was asking to add "macro", I wanted to add it to acceptable values `average in ["micro", "macro", "weighted", "samples"]`. I agree that `average=True` is the same as `average="macro"`. But it is not bad. We are gradually deprecating previous api to the new one with average as str similar to scikit-learn.
By the way, we have to add `average=None` as well which maps to `average=False`. 


What do you think about trying to match Scikit-learn API for average arg ?

Reply:
Useful for users familiar with Scikit. If so, shall I add `pos_label` as well? (for `average='binary'`) As I said above, `pos_label` is another parameter which is used to determine precision/recall of which class be returned.

Reply:
Yes, it can be useful, but let's do that in another PR, not here

Reply:
On the other hand , it decreases our api's elegance.